### PR TITLE
feat: Add a feature to change country display format in form

### PR DESCRIPTION
**Pull request by nicolassing**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,6 +162,12 @@ This produces the preferred choices of 'Jersey' and 'United Kingdom', and regula
 By default the `country_choices` is empty, which means all countries are included, as is `preferred_country_choices`.
 The option `country_placeholder` can be specified to create a placeholder option on above the whole list.
 
+The option `country_display_type` can be specified to change the country dropdown label format. There is three formats availables :
```
Comment (Nek-) [inline README.md:165]:
```suggestion
The option `country_display_type` can be specified to change the country dropdown label format. There are two formats available :
```

You suggest only 2 formatting for the country name, isn't it? 🤔

File: src/Form/Type/PhoneNumberType.php [hunk 2]
```diff
@@ -128,6 +132,7 @@ public function configureOptions(OptionsResolver $resolver)
             'by_reference' => false,
             'error_bubbling' => false,
             'country_choices' => [],
+            'country_display_type' => self::DISPLAY_COUNTRY_FULL,
```
Comment (Nek-) [inline src/Form/Type/PhoneNumberType.php:135]: This looks like a break, we currently use short format, isn't it?

File: src/Form/Type/PhoneNumberType.php [hunk 3]
```diff
@@ -61,12 +64,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
             $countryChoices = [];
 
-            foreach (Countries::getNames() as $region => $name) {
-                if (false === isset($countries[$region])) {
+            foreach ($countries as $regionCode => $countryCode) {
```
Comment (Nek-) [inline src/Form/Type/PhoneNumberType.php:67]: Why this change? It's not equivalent to what was here before. Therefore it potentially breaks the compatibility.

File: README.md (hunk 1 above)
Comment (nicolassing) [reply README.md:165]: In the first place, i've planned to have 3 formats. But the third, only the country code (eg. +33) does not work because a region code can be used by many regionCode.

File: src/Form/Type/PhoneNumberType.php (hunk 2 above)
Comment (nicolassing) [reply src/Form/Type/PhoneNumberType.php:135]: nop, currently we use "France (+33)" the "FULL" format

File: src/Form/Type/PhoneNumberType.php (hunk 3 above)
Comment (nicolassing) [reply src/Form/Type/PhoneNumberType.php:67]: The results is almost the same. I made this change to keep the order given in the `countries` configuration entry.

**Comment by nicolassing** [comment]
Hi ! 
If someone want to discuss about this PR, let me know :)

File: src/Form/Type/PhoneNumberType.php (hunk 3 above)
Comment (maxhelias) [reply src/Form/Type/PhoneNumberType.php:67]: If I'm not mistaken, the performances are not the same this way because before the data was loaded and browsed only once in the buffer of the component while now we have it as many times as countries support x2 (exists & getName)

Comment (nicolassing) [reply src/Form/Type/PhoneNumberType.php:67]: You're right ! Not sure its a big deal. Let me check :)

Comment (maxhelias) [reply src/Form/Type/PhoneNumberType.php:67]: I would not be so convinced. I would like to have a verification on this before

Comment (Nek-) [reply src/Form/Type/PhoneNumberType.php:67]: The performance overhead is actually small. But I'm surprised there's indeed a topic here. That said, a solution can be quickly found here:

```php
Countries::exists($regionCode)
```

could be changed easily with

```php
$intlCountries = Countries::getNames();
// and later
if (!isset($intlCountries[$regionCode])
```

File: src/Form/Type/PhoneNumberType.php (hunk 2 above)
Comment (Nek-) [reply src/Form/Type/PhoneNumberType.php:135]: my bad

**Comment by Nek-** [comment]
Hello @nicolassing sorry for not taking the time to have a look, and thank you very much for your patience. This feature is nice and can be merged.

Can you add the (little) optimization I suggested?

**Review by Nek- — APPROVED** [review]

**Comment by Nek-** [comment]
Thank you very much @nicolassing !
